llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -2080,6 +2080,11 @@ size_t SYCLLowerESIMDPass::runOnFunction(Function &F,
       // process ESIMD builtins that go through special handling instead of
       // the translation procedure
 
+      if (Name.starts_with("__esimd_slm_alloc") ||
+          Name.starts_with("__esimd_slm_free")) {
+        continue;
```

Comment:
Maybe add a comment saying they will be lowered later in LowerESIMDSlmReservationCalls